### PR TITLE
refactor: create separate schema for Ajv validation

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
How it's working now:
It takes a user-defined schema, adds necessary adjustments for Ajv (now it's only for date type), and then works with modified schema.

How it will work:
It will clone a user-defined schema, modify it and pass it to the Ajv. Serialization will work with the original schema. The only requirement we must follow is that the schema must not change its structure during the modification. In other words, the JSON pointers must match in two schemas, but the schemas themselves (type, format, keywords, ... etc.) may differ.

Motivation:
It's too complicated to maintain one syntax for serialization and validation rules. Our code depends on Ajv validation syntax.

Depends on #507

### CODE REVIEW

**Comment by Uzlopak** [comment]
It makes sense to me.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ivan-tymoshenko** [comment]
There are two updates.
1. I moved all Ajv logic to the Validator class. The Validator is a class Adapter. It has two public methods `addSchema` and `validate`. It should validate schema according to our rules. It encapsulates all Ajv validation logic in one place. 
2. There is a small change in API. If you call the build method in the debug mode, one of the return parameters would be the Validator instance instead of the Ajv instance.

File: ref-resolver.js [hunk 1]
```diff
@@ -11,8 +11,10 @@ class RefResolver {
     if (schema.$id !== undefined && schema.$id.charAt(0) !== '#') {
       schemaId = schema.$id
     }
-    this.insertSchemaBySchemaId(schema, schemaId)
-    this.insertSchemaSubschemas(schema, schemaId)
+    if (this.getSchema(schemaId) === undefined) {
```
Comment (climba03003) [inline ref-resolver.js:14]: Is this changes itself self contained?
Do we need a test case for it?

**Review by climba03003 — APPROVED** [review]
The code itself look good to me and the changes on `debug` output is breaking.

@ivan-tymoshenko
You are the one focusing on this package. Can I ask how many breaking changes are you expected in the near future? I am considering should we group a bunch of them before another major.

**Comment by ivan-tymoshenko** [comment]
@climba03003 I can leave the Ajv instance and remove it in the next major release.

File: ref-resolver.js (hunk 1 above)
Comment (ivan-tymoshenko) [reply ref-resolver.js:14]: I didn't add this check. I just moved it inside the class.
https://github.com/fastify/fast-json-stringify/blob/b0bb1a699e5faed3657ad8c1d665920e29304be5/index.js#L113

**Pull request merged**